src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -0,0 +1,512 @@
+        List<String> getIncludes();
+
+        @Config("dedup")
+        @ConfigDefault("true")
+        boolean getDedup();
+
+        @Config("app_marketplace_integration_name")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceIntegrationName();
+
+        @Config("app_marketplace_org_id")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceOrgId();
+
+        @Config("app_marketplace_app_id")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceAppId();
+
+        @Config("columns")
+        SchemaConfig getColumns();
+    }
+
+    private static final Logger logger = Exec.getLogger(ZendeskInputPlugin.class);
+
+    private ZendeskSupportAPIService zendeskSupportAPIService;
+
+    @Override
+    public ConfigDiff transaction(final ConfigSource config, final Control control)
+    {
+        final PluginTask task = config.loadConfig(PluginTask.class);
+        final Schema schema = task.getColumns().toSchema();
+        int taskCount = 1;
+        if (!ZendeskUtils.isSupportIncremental(task.getTarget())) {
+            final JsonNode result = getZendeskSupportAPIService(task).getData("", 0, false, 0);
+            if (result.get(ZendeskConstants.Field.COUNT) != null
+                    && result.get(ZendeskConstants.Field.COUNT).isInt()) {
+                taskCount = ZendeskUtils.numberToSplitWithHintingInTask(result.get(ZendeskConstants.Field.COUNT).asInt());
+            }
+        }
+        return resume(task.dump(), schema, taskCount, control);
+    }
+
+    @Override
+    public ConfigDiff resume(final TaskSource taskSource, final Schema schema, final int taskCount, final Control control)
+    {
+        final PluginTask task = taskSource.loadTask(PluginTask.class);
+        final List<TaskReport> taskReports = control.run(taskSource, schema, taskCount);
+
+        return this.buildConfigDiff(task, taskReports);
+    }
+
+    @Override
+    public void cleanup(final TaskSource taskSource, final Schema schema, final int taskCount, final List<TaskReport> successTaskReports)
+    {
+    }
+
+    @Override
+    public TaskReport run(final TaskSource taskSource, final Schema schema, final int taskIndex, final PageOutput output)
+    {
+        final PluginTask task = taskSource.loadTask(PluginTask.class);
+        ZendeskValidatorUtils.validateInputTask(task, getZendeskSupportAPIService(task));
+        try (final PageBuilder pageBuilder = getPageBuilder(schema, output)) {
+            final TaskReport taskReport = ingestServiceData(task, taskIndex, schema, pageBuilder);
+            pageBuilder.finish();
+            return taskReport;
+        }
+    }
+
+    @Override
+    public ConfigDiff guess(final ConfigSource config)
+    {
+        config.set("columns", new ObjectMapper().createArrayNode());
+        final PluginTask task = config.loadConfig(PluginTask.class);
+        ZendeskValidatorUtils.validateInputTask(task, getZendeskSupportAPIService(task));
```

Comment:
same as above.

Reply:
Because we already move `validateInputTask` to transaction, it is safe to remove in `guess`, or is there any reason it needs to be in both?

Reply:
Keep the code like old logic.
https://github.com/treasure-data/embulk-input-zendesk/blob/master/lib/embulk/input/zendesk/plugin.rb#L42